src/hooks/useUsers.ts

```diff
@@ -2,26 +2,35 @@ import { useEffect, useState } from 'react';
 import { getUsers } from '../queries/users';
 import { IUser } from '../types';
 
-const useUsers = (searchQuery?: string): IUser[] => {
+const useUsers = (
+  searchQuery?: string,
+  numberPerPage?: number,
+  offset?: number,
+): { noMoreData: boolean; loading: boolean; users: IUser[] } => {
   const [users, setUsers] = useState<IUser[]>([]);
+  const [noMoreData, setNoMoreData] = useState(false);
```

Comment:
Booleans must globally use the 'is' prefix: isSet, isVisible, isFinished, isFound, isOpen
There are a few possible alternatives that make more sense in some cases:

hasLicense;
canEvaluate;
shouldAbort;
useForFirstPartyData;

Here we can rename to hasMoreData